### PR TITLE
Changing the grid2d dip.

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1002 - dip and rotation**
Now, the crop grid 2d take account the dip

The vertical is now correlated with the dip

### CODE REVIEW

File: tests/grid_2d_test.py [hunk 1]
```diff
@@ -151,3 +153,43 @@ def test_grid2d_to_geoimage(tmp_path):
 
         with pytest.raises(TypeError, match="The dtype of the keys must be"):
             converter.key_to_data(grid, [0, 1])
+
+
+def test_grid2d_dip(tmp_path):
+    workspace = Workspace(tmp_path / r"geo_image_test.geoh5")
+
+    _ = Grid2D.create(
```
Comment (domfournier) [inline tests/grid_2d_test.py:161]: Never used?

File: tests/grid_2d_test.py [hunk 2]
```diff
@@ -151,3 +153,43 @@ def test_grid2d_to_geoimage(tmp_path):
...
+    grid1.rotation = 66
+
+    grid2 = grid.copy()
+
+    grid2.dip = 90
+
+    grid3 = grid.copy()
+
+    grid3.rotation = 66
+    grid3.dip = 44
```
Comment (domfournier) [inline tests/grid_2d_test.py:195]: What are we testing here?

**Review by domfournier — COMMENTED** [review]
Just wondering what's happening ion the grid_2d_test? Rest looks good.

File: tests/grid_2d_test.py (hunk 2 above)
Comment (MatthieuCMira) [reply tests/grid_2d_test.py:195]: Ah, I was just testing the dip setter in GA. 
I'm removing it now!

**Pull request merged**